### PR TITLE
swift-t: New release packages for 1.4.3

### PR DESCRIPTION
**Pull request by jmjwozniak**

### CODE REVIEW

**Comment by jmjwozniak** [comment]
This is a reboot of #11376 .  
Please prioritize this PR.  The previous PR failed because Spack changed while this queued for review.

File: var/spack/repos/builtin/packages/turbine/package.py [hunk 1]
```diff
@@ -15,8 +15,8 @@ class Turbine(AutotoolsPackage):
     git      = "https://github.com/swift-lang/swift-t.git"
     configure_directory = 'turbine/code'
 
-    version('develop', branch='master')
-    version('1.2.3', 'f2e393c292c4248b4e77a19f8272ae88')
+    version('master', branch='master')
+    version('1.2.3', '028f6f4f5041e5ffbbf3ab8008e3b781')
```
[the remark this answers is not in the document]
Comment (jmjwozniak) [reply var/spack/repos/builtin/packages/turbine/package.py]: I am the developer, this checksum is correct.

[the remark this answers is not in the document]
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/turbine/package.py]: In that case, you can add:
```python
maintainers = ['jmjwozniak']
```
to these packages if you want. That way we can notify you if someone makes a change or has a problem installing the package.

**Comment by jmjwozniak** [comment]
We like the ability to install from Git or from tarballs.  This is great so users that are running without Spack can use the same tarballs that the Spack installations are using.

**Comment by citibeth** [comment]
Unfortunately, this PR as it stands does not give the ability to install
any version other than master from git.

**Comment by citibeth** [comment]
@adamjstewart wouldn’t it be great if these kinds of smarts were built into
AutotoolsPackage?

**Comment by jmjwozniak** [comment]
Right- is that required?  Our main use case is the tarballs.  We only recently added the git option, this is just for quick fixes/features from git for particular advanced users.

**Comment by adamjstewart** [comment]
Spack doesn't currently have a way to add dependencies for all git versions, so I don't think that should be required to merge this PR.

**Comment by jmjwozniak** [comment]
Ok, is there anything else we need to do?

**Comment by citibeth** [comment]
@adamjstewart I think you misunderstood.  My question to you wasn't about this PR; but rather, asking for your assessment on how easy/hard it would be to make `AutotoolsPackage` smarter?  It could be smarter in two ways:

1. It should know how to add the `automake`, etc. dependencies --- which currently, we manually add on every package that needs it.

2. It should be smart enough to know (by default) that, unless told otherwise, non-tarball downloads require the extra dependencies and steps.

WIth these two features in place, this PR would "just work", and be more flexible, without all the extra `when='master'` stuff.  So my question is: *How easy/hard would this be to add to `AutotoolsPackage`, and what technical aspects do you think it would entail??*

@alalazo CC'd because he's done a lot on per-build-tools package base classes.

**Comment by adamjstewart** [comment]
I think it would be easy to add something like:
```python
depends_on('automake', when='@develop,master')
depends_on('autoconf', when='@develop,master')
depends_on('libtool', when='@develop,master')
depends_on('m4', when='@develop,master')
```
to the `AutotoolsPackage` base class. But adding dependencies for all non-tarball downloads would be much more challenging, as Spack currently doesn't forward that information to the spec, so it can't be used in a `when` clause.

**Pull request merged**